Intern/rayx-core/src/Beamline/DatFile.cpp

```diff
@@ -9,46 +9,53 @@
 
 namespace RAYX {
 bool DatFile::load(const std::filesystem::path& filename, DatFile* out) {
-    std::ifstream s(filename);
-    RAYX_VERB << filename;
-    std::string line;
-
-    // line 1
-    std::getline(s, out->m_title);
-
-    // line 2
-    std::getline(s, line);
-#if defined(WIN32)
-    if (sscanf_s(line.c_str(), "%u %le %le %le", &out->m_lineCount, &out->m_start, &out->m_end, &out->m_step) != 4) {
-#else
-    if (sscanf(line.c_str(), "%u %le %le %le", &out->m_lineCount, &out->m_start, &out->m_end, &out->m_step) != 4) {
-#endif
-        RAYX_EXIT << "Failed to parse DatFile \"" << filename << "\", at line 2: \"" << line << "\"";
-        return false;
-    }
-    out->m_Lines.reserve(out->m_lineCount);
-
-    // line 3..EOF
-    out->m_weightSum = 0;
-    for (uint32_t lineidx = 3; std::getline(s, line); lineidx++) {
-        if (line.empty()) {
-            continue;
+    try
+    {   
+        std::ifstream s(filename);
+        RAYX_VERB << filename;
+        std::string line;
+
+        // line 1
+        std::getline(s, out->m_title);
+
+        // line 2
+        std::getline(s, line);
+    #if defined(WIN32)
+        if (sscanf_s(line.c_str(), "%u %le %le %le", &out->m_lineCount, &out->m_start, &out->m_end, &out->m_step) != 4) {
+    #else
+        if (sscanf(line.c_str(), "%u %le %le %le", &out->m_lineCount, &out->m_start, &out->m_end, &out->m_step) != 4) {
+    #endif
+            RAYX_D_ERR << "Failed to parse DatFile \"" << filename << "\", at line 2: \"" << line << "\"";
+            return false;
         }
+        out->m_Lines.reserve(out->m_lineCount);
 
-        DatEntry e{};
-#if defined(WIN32)
-        if (sscanf_s(line.c_str(), "%le %le", &e.m_energy, &e.m_weight) != 2) {
-#else
-        if (sscanf(line.c_str(), "%le %le", &e.m_energy, &e.m_weight) != 2) {
-#endif
-            RAYX_EXIT << "Failed to parse DatFile \"" << filename << "\", at line " << lineidx << ": \"" << line << "\"";
-            return false;
+        // line 3..EOF
+        out->m_weightSum = 0;
+        for (uint32_t lineidx = 3; std::getline(s, line); lineidx++) {
+            if (line.empty()) {
+                continue;
+            }
+
+            DatEntry e{};
+    #if defined(WIN32)
+            if (sscanf_s(line.c_str(), "%le %le", &e.m_energy, &e.m_weight) != 2) {
+    #else
+            if (sscanf(line.c_str(), "%le %le", &e.m_energy, &e.m_weight) != 2) {
+    #endif
+                RAYX_D_ERR << "Failed to parse DatFile \"" << filename << "\", at line " << lineidx << ": \"" << line << "\"";
+                return false;
+            }
+            out->m_Lines.push_back(e);
+            out->m_weightSum += e.m_weight;
         }
-        out->m_Lines.push_back(e);
-        out->m_weightSum += e.m_weight;
-    }
 
-    return true;
+        return true;
+
+    } catch (const std::exception& e) {
+        RAYX_D_ERR << "Exception caught while loading DatFile \"" << filename << "\": " << e.what();
```

Comment:
On further review I believe that lazy loading the energy file during the compile call creates quite a few problems. A broken DatFile path in the rml file should already be found (but not handled) during the importBeamline(...) function call. rayx could then exit with the error message, that the file could not be found at the given path or it could not be parsed. rayx-ui could prompt the user with a pop-up with a similar/the same error message